tests/unit/test_utils.py

```diff
@@ -1,9 +1,236 @@
 import re
 
+import pyam
+import numpy as np
+import pandas as pd
 import pytest
 
-from silicone.utils import _get_unit_of_variable
+from silicone.utils import _get_unit_of_variable, find_matching_scenarios, make_interpolator
 
+_mc = "model_c"
+_sa = "scen_a"
+_sb = "scen_b"
+_eco2 = "Emissions|CO2"
+_gtc = "Gt C/yr"
+_ech4 = "Emissions|CH4"
+_mtch4 = "Mt CH4/yr"
+_msrvu = ["model", "scenario", "region", "variable", "unit"]
+simple_df = pd.DataFrame(
+        [
+            [_mc, _sa, "World", _eco2, _gtc, 0, 1000, 5000],
+            [_mc, _sb, "World", _eco2, _gtc, 1, 1000, 5000],
+            [_mc, _sa, "World", _ech4, _mtch4, 0, 300, 500],
+            [_mc, _sb, "World", _ech4, _mtch4, 1, 300, 500],
+        ],
+        columns=_msrvu + [2010, 2030, 2050],
+    )
+simple_df = pyam.IamDataFrame(simple_df)
+
+
+def test_find_matching_scenarios():
+    variable_leaders = ["Emissions|CO2"]
+    variable_follower = "Emissions|CH4"
+    half_simple_df = simple_df.filter(scenario="scen_a")
+    scenarios = find_matching_scenarios(
+        simple_df,
+        half_simple_df,
+        variable_follower,
+        variable_leaders,
+        ["scen_a", "scen_b"],
+    )
+    assert scenarios == ("*", "scen_a")
+    half_simple_df.data["value"].loc[0] = 0.49
+    scenarios = find_matching_scenarios(
+        simple_df,
+        half_simple_df,
+        variable_follower,
+        variable_leaders,
+        ["scen_a", "scen_b"],
+    )
+    assert scenarios == ("*", "scen_a")
+    half_simple_df.data["value"].loc[0] = 0.51
+    scenarios = find_matching_scenarios(
+        simple_df,
+        half_simple_df,
+        variable_follower,
+        variable_leaders,
+        ["scen_a", "scen_b"],
+    )
+    assert scenarios == ("*", "scen_b")
+
+
+def test_find_matching_scenarios_no_data_for_time():
+    variable_leaders = ["Emissions|CO2"]
+    variable_follower = "Emissions|CH4"
+    time_col = simple_df.time_col
+    half_simple_df = simple_df.filter(scenario="scen_a")
+    half_simple_df.data[time_col].loc[0] = 0
+    with pytest.raises(ValueError):
+        find_matching_scenarios(
+            simple_df,
+            half_simple_df,
+            variable_follower,
+            variable_leaders,
+            ["scen_a", "scen_b"],
+        )
+
+
+def test_find_matching_scenarios_use_change_instead_of_absolute():
+    # In this case, we will ignore any offset
+    variable_leaders = ["Emissions|CO2"]
+    variable_follower = "Emissions|CH4"
+    half_simple_df = simple_df.filter(scenario="scen_a")
+    half_simple_df.data["value"] = half_simple_df.data["value"] - 10000
+    scenarios = find_matching_scenarios(
+        simple_df,
+        half_simple_df,
+        variable_follower,
+        variable_leaders,
+        ["scen_a", "scen_b"],
+        use_change_not_abs=True,
```

Comment:
if you use `use_change_not_abs=False` here, do you `scen_b` back?

Reply:
Actually not, I will change the plus sign to a minus